modelopt/onnx/autocast/precisionconverter.py

```diff
@@ -1050,3 +1071,27 @@ def _is_foldable_constant_cast_pattern(self, node: onnx.NodeProto) -> bool:
             get_consumer_nodes = utils.get_consumer_nodes(self.model, const_producer.output[0])
             return len(get_consumer_nodes) == 1 and get_consumer_nodes[0] == node
         return False
+
+    def _should_skip_low_precision_input_conversion(
+        self, node: onnx.NodeProto, input_name: str
+    ) -> bool:
+        """Check if the input should be skipped for low precision conversion.
+
+        This is used for nodes that have inputs that MUST remain in FP32.
+        """
+        assert isinstance(node, onnx.NodeProto), f"node must be an onnx.NodeProto, got {type(node)}"
+        assert isinstance(input_name, str), f"input_name must be a string, got {type(input_name)}"
+        if node.op_type == "Resize":
+            if input_name not in node.input:
+                raise KeyError(
+                    f"Input {input_name} not found in node {node.name} input, not expected!"
+                )
+            # Figure out the index of the input in the node input
+            inputs_lst = list(node.input)
+            input_index = inputs_lst.index(input_name)
+            # The second input does not support bfloat16, so leave it in FP32.
+            # The third input of Resize must remain in FP32.
+            # Ref: https://onnx.ai/onnx/operators/onnx__Resize.html#inputs
+            return input_index in {1, 2}
```

Comment:
We can check if we're converting to fp16/bf16 and provide a more accurate fix:
```suggestion
            # The second input does not support bfloat16, so leave it in FP32.
            # The third input of Resize must remain in FP32.
            # Ref: https://onnx.ai/onnx/operators/onnx__Resize.html#inputs
            return input_index in {1, 2} if self.low_precision_type == "fp16" else {1}
```

Reply:
Addressed using your other comment